src/GitHub.Api/GraphQLClient.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Runtime.Caching;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHub.Extensions;
+using Microsoft.VisualStudio.Threading;
+using Octokit.GraphQL;
+using Octokit.GraphQL.Core;
+
+namespace GitHub.Api
+{
+    public class GraphQLClient : IGraphQLClient
+    {
+        public static readonly TimeSpan DefaultCacheDuration = TimeSpan.FromHours(8);
+        readonly IConnection connection;
+        readonly FileCache cache;
+
+        public GraphQLClient(
+            IConnection connection,
+            FileCache cache)
+        {
+            this.connection = connection;
+            this.cache = cache;
+        }
+
+        public async Task ClearCache(string regionName)
+        {
+            // Switch to background thread because FileCache does not provide an async API.
+            await TaskScheduler.Default;
```

Comment:
I wonder if we should avoid a `Microsoft.VisualStudio.Threading` dependency simply returning `Task.Run(() => cache.ClearRegion(GetFullRegionName(regionName)))`?